network/dag/interface.go

```diff
@@ -133,16 +135,17 @@ type Visitor func(ctx context.Context, transaction Transaction) bool
 // PayloadStore defines the interface for types that store and read transaction payloads.
 type PayloadStore interface {
 	PayloadReader
-	PayloadWriter
-	// ReadManyPayloads allows the caller read many payloads in an optimized fashion.
+	// WritePayload writes contents for the specified payload, identified by the given hash.
+	WritePayload(ctx context.Context, payloadHash hash.SHA256Hash, data []byte) error
```

Comment:
the inconsistency we now have with PayloadWriter.WritePayload reeks like bad design? PayloadStore composes from PayloadReader and not PayloadWriter? Maybe just pass in the TX here as well?

Reply:
The bad design imho was that the business logic and db impl used the same interface. A renaming of interfaces/method to make this clearer could be a solution.

Reply:
Probably, but we can also leave it to be refactored part of the storage layer?